neqo-transport/src/cc/classic_cc.rs

```diff
@@ -764,4 +818,133 @@ mod tests {
         let lost = make_lost(&[PERSISTENT_CONG_THRESH + 2, 1]);
         assert!(!persistent_congestion_by_pto(0, 0, &lost));
     }
+
+    #[test]
+    fn app_limited_slow_start() {
+        let mut cc = ClassicCongestionControl::new(NewReno::default());
+
+        for i in 0..CWND_INITIAL_PKTS {
+            let p = SentPacket::new(
+                PacketType::Short,
+                u64::try_from(i).unwrap(), // pn
+                now(),                     // time sent
+                true,                      // ack eliciting
+                Rc::default(),             // tokens
+                MAX_DATAGRAM_SIZE,         // size
+            );
+            cc.on_packet_sent(&p);
+        }
+        assert_eq!(cc.bytes_in_flight(), CWND_INITIAL);
+
+        for i in 0..4 {
```

Comment:
I think that we need a constant for the count here.  And an assertion that `LESS_THAN_CWND_PKTS < CWND_INITIAL_PKTS / 2`.